src/CharTable.jl

```diff
@@ -3,6 +3,8 @@ export genchartab, greenfuntab
 abstract type Table{T} end
 abstract type AbstractGenericCharacter{T} end
 
+# TODO The type parameter T is not needed for CharTable and GenericCharacter
```

Comment:
But what do you mean by that and what is stopping you from removing it? As I see it, there still is `chars::Vector{<:AbstractGenericCharacter{T}}` so it is "needed"? But perhaps `AbstractGenericCharacter` (and `Table`) can just drop it?

Reply:
Of course there is also `SimpleCharTable{T}` but isn't `T` essentially a constant there, too? Or can it actually take on different values?

Reply:
For `SimpleCharTable{T}` it is either `PolyRingElem{QQFieldElem}` or `PolyRingElem{AbsSimpleNumFieldElem}`. For example the Green Functions of `2B2` need a square root of three.

Reply:
Of course `SimpleCharTable` could also retain its type parameter (I see no harm in it), but that doesn't mean the abstract base type needs it? I looked at how those are used in current master and the type parameter on the abstract type was useless in all cases I saw (I may have missed some, though). E.g.:
```julia
function omega!(t::Table{T}, char::Int64) where T <: NfPoly
	if char > chartypes(t)
		throw(DomainError(char, "Character type is out of range."))
	end
	push!(t.chars, omega(t[char]))
	return length(t.chars)
end
```
could just as well be
```julia
function omega!(t::Table, char::Int64)
	if char > chartypes(t)
		throw(DomainError(char, "Character type is out of range."))
	end
	push!(t.chars, omega(t[char]))
	return length(t.chars)
end
```
(though of course we actually want to get rid of `omega!`)

Reply:
> Of course SimpleCharTable could also retain its type parameter (I see no harm in it)

Ok, I'll do that.

The change of `omega!` is actually what I've done in this PR already since `T <: NfPoly` doesn't hold anymore.

Reply:
>> Of course SimpleCharTable could also retain its type parameter (I see no harm in it),...

> Ok, I'll do that.

I've looked into it but I actually don't know how you mean that. The only way to get rid of the dependence on `T` I can think of is replacing all occurrences of `T` by `NfPoly` or `<:NfPoly`.

Reply:
Ok, I got it now. It's working.